### ISSUE TITLE
Node properties are null in system.peers for node that was decommissioned and removed

### ISSUE THREAD

**Issue by juliayakovlev** [scylladb/scylla#6397]
*Installation details*
Scylla version (or git commit hash): 4.0.rc3-0.20200501.eee4c00e29
Cluster size: 9 nodes (multidc)
OS (RHEL/CentOS/Ubuntu/AWS AMI): ami-016482914844d8a49,  ami-08eeb1a787e407ef9, ami-0c6075ec8f222315b
 
Found "null" value in node properties in system.peers for node that was decommissioned and removed.
Maybe as issue [#4652](https://github.com/scylladb/scylla/issues/4652)

This is the flow before the error.

1. ```2020-05-06 13:18:12,623``` - Started random_disrupt_method grow_shrink_cluster
    During this nemesis node node 8  with IP 54.165.104.245 was decommissioned and removed:
```
2020-05-06T13:40:48+00:00  longevity-counters-multidc-refactor-db-node-42faa265-8 !INFO    | scylla: [shard 0] gossip - My status = LEFT
```

2. ```2020-05-06 14:45:19,090``` - Started random_disrupt_method restart_then_repair_node on node 9 (longevity-counters-multidc-refactor-db-node-42faa265-9, IP: [3.82.216.102 | 172.30.0.246])
    ```2020-05-06 14:50:51,086``` - Finished

3. ```2020-05-06 15:31:20,395``` - Started random_disrupt_method stop_wait_start_scylla_server on node 9
    ```2020-05-06 15:37:35,009``` - Finished

Error: SYSTEM.PEERS table  on node 9 keeps row about decommissioned node 8 with IP 54.165.104.245 but value in every column became "null":
```
'54.165.104.245 |        null |           null | null |            null |           null |         null"
```

Property for node 8 is not null in system.peers on all other nodes.

Should the info about decommissioned node be kept in the system.peers?

**Comment by bentsi** [comment]
happened to me too, after decommission nemesis in alternator multi-dc scenario. Pasted details in #4652

**Comment by slivne** [comment]
@asias  any explanation on this ?

**Comment by asias** [comment]
> Error: SYSTEM.PEERS table on node 9 keeps row about decommissioned node 8 with IP 54.165.104.245 but value in every column became "null":
> 
> ```
> '54.165.104.245 |        null |           null | null |            null |           null |         null"
> ```
> 
> Property for node 8 is not null in system.peers on all other nodes.

Columns for node 8 in system.peers was not fully shown here. e.g., tokens.  I think you only selected part of the columns in the sct. We need to know all the columns. Can you please modify the sct test so that when we hit it again we know the full info in system.peers? @juliayakovlev .

It is similar for @bentsi 's case [here](https://github.com/scylladb/scylla/issues/4652#issuecomment-627664189)

**Comment by asias** [comment]
> * `2020-05-06 14:45:19,090` - Started random_disrupt_method restart_then_repair_node on node 9 (longevity-counters-multidc-refactor-db-node-42faa265-9, IP: [3.82.216.102 | 172.30.0.246])
>   `2020-05-06 14:50:51,086` - Finished
> 
> * `2020-05-06 15:31:20,395` - Started random_disrupt_method stop_wait_start_scylla_server on node 9
>   `2020-05-06 15:37:35,009` - Finished


@juliayakovlev can you please help me understand what are those operations  are doing?

According log of node 9, node9 restarted only once around 2020-05-06T15:31:31. No restart during the period of 2020-05-06 14:45:19 to 2020-05-06 14:50:51 as mentioned above for the random_disrupt_method restart_then_repair_node. What was going on? What did random_disrupt_method stop_wait_start_scylla_server do?

1) 2020-05-06T13:01:15+00:00  longevity-counters-multidc-refactor-db-node-42faa265-9 !INFO    | scylla: Scylla version 4.0.rc3-0.20200501.eee4c00e29 with build-id 0fe095b80ee54c7f47a688aaac0c6bb835118ddc starting ...

2) 2020-05-06T15:31:31+00:00  longevity-counters-multidc-refactor-db-node-42faa265-9 !INFO    | scylla: [shard 0] init - Scylla version 4.0.rc3-0.20200501.eee4c00e29 shutdown complete.

3) 2020-05-06T15:36:33+00:00  longevity-counters-multidc-refactor-db-node-42faa265-9 !INFO    | scylla: Scylla version 4.0.rc3-0.20200501.eee4c00e29 with build-id 0fe095b80ee54c7f47a688aaac0c6bb835118ddc starting ...

**Comment by juliayakovlev** [comment]
> > Error: SYSTEM.PEERS table on node 9 keeps row about decommissioned node 8 with IP 54.165.104.245 but value in every column became "null":
> > ```
> > '54.165.104.245 |        null |           null | null |            null |           null |         null"
> > ```
> > 
> > 
> > Property for node 8 is not null in system.peers on all other nodes.
> 
> Columns for node 8 in system.peers was not fully shown here. e.g., tokens. I think you only selected part of the columns in the sct. We need to know all the columns. Can you please modify the sct test so that when we hit it again we know the full info in system.peers? @juliayakovlev .
> 

Sure, I will add it

> It is similar for @bentsi 's case [here](https://github.com/scylladb/scylla/issues/4652#issuecomment-627664189)

**Comment by juliayakovlev** [comment]
> > * `2020-05-06 14:45:19,090` - Started random_disrupt_method restart_then_repair_node on node 9 (longevity-counters-multidc-refactor-db-node-42faa265-9, IP: [3.82.216.102 | 172.30.0.246])
> >   `2020-05-06 14:50:51,086` - Finished
> > * `2020-05-06 15:31:20,395` - Started random_disrupt_method stop_wait_start_scylla_server on node 9
> >   `2020-05-06 15:37:35,009` - Finished
> 
> @juliayakovlev can you please help me understand what are those operations are doing?
> 
> According log of node 9, node9 restarted only once around 2020-05-06T15:31:31. No restart during the period of 2020-05-06 14:45:19 to 2020-05-06 14:50:51 as mentioned above for the random_disrupt_method restart_then_repair_node. What was going on? 

Yes, real restart wasn't run because of the spot instances are used in the test:
```
< t:2020-05-06 14:45:19,103 f:cluster_aws.py  l:589  c:sdcm.cluster_aws     p:DEBUG > target node is spot instance, impossible to stop this instance, skipping the restart
``` 

> What did random_disrupt_method stop_wait_start_scylla_server do?

This nemesis run:
1. ```systemctl stop scylla-server```
2. sleep 10 min
3. ```systemctl start scylla-server```

> 
> 1. 2020-05-06T13:01:15+00:00  longevity-counters-multidc-refactor-db-node-42faa265-9 !INFO    | scylla: Scylla version 4.0.rc3-0.20200501.eee4c00e29 with build-id 0fe095b80ee54c7f47a688aaac0c6bb835118ddc starting ...
> 2. 2020-05-06T15:31:31+00:00  longevity-counters-multidc-refactor-db-node-42faa265-9 !INFO    | scylla: [shard 0] init - Scylla version 4.0.rc3-0.20200501.eee4c00e29 shutdown complete.
> 3. 2020-05-06T15:36:33+00:00  longevity-counters-multidc-refactor-db-node-42faa265-9 !INFO    | scylla: Scylla version 4.0.rc3-0.20200501.eee4c00e29 with build-id 0fe095b80ee54c7f47a688aaac0c6bb835118ddc starting ...

**Comment by asias** [comment]
Thanks. One more thing: at which point the test found the null items for 54.165.104.245? Was systems.peer checked both before and after the restart of node 9?

**Comment by juliayakovlev** [comment]
> Thanks. One more thing: at which point the test found the null items for 54.165.104.245? Was systems.peer checked both before and after the restart of node 9?

This validation runs before and after every nemesis. It means that before stop_wait_start_scylla_server nemesis the properties in the system.peers were ok

**Comment by asias** [comment]
> > Thanks. One more thing: at which point the test found the null items for 54.165.104.245? Was systems.peer checked both before and after the restart of node 9?
> 
> This validation runs before and after every nemesis. It means that before stop_wait_start_scylla_server nemesis the properties in the system.peers were ok

Good. This means the null entries were created only after the restart of node 9. I will dig more.

**Comment by slivne** [comment]
@asias do you patches fix this issue ?

**Comment by denesb** [comment]
Closing as stale.
